lib/model/message_list.dart

```diff
@@ -528,12 +567,29 @@ class MessageListView with ChangeNotifier, _MessageSequence {
 
       _insertAllMessages(0, fetchedMessages);
       _haveOldest = result.foundOldest;
+      _fetchOlderCooldownBackoffMachine = null;
     } finally {
-      if (this.generation == generation) {
-        _fetchingOlder = false;
-        _updateEndMarkers();
-        notifyListeners();
+      if (this.generation != generation) {
+        // We need the finally block always clean up regardless of errors
+        // occured in the try block, and returning early here is necessary
+        // if such cleanup must be skipped, as the fetch is considered stale.
+        // ignore: control_flow_in_finally
+        return;
```

Comment:
```dart
      if (this.generation != generation) {
        // We need the finally block always clean up regardless of errors
        // occured in the try block, and returning early here is necessary
        // if such cleanup must be skipped, as the fetch is considered stale.
        // ignore: control_flow_in_finally
        return;
```

This explanation doesn't make sense to me. The cleanup "always" needs to be done…but there are times when it "must be skipped"? Your reasoning might be correct but I don't understand it yet.

Reply:
As a reminder, the task is to explain why the problems flagged by `control_flow_in_finally` either don't exist or are acceptable.

Reply:
How about:
```
        // This lint rule is more helpful for flagging confusing uses of
        // control flow keywords, such as `break` in a block nested in a loop,
        // or `return` a value in a finally block when a value would have been
        // returned in the `try` block.  Returning early to skip the cleanup
        // is none of these.
        // ignore: control_flow_in_finally
```

I was also tempted to refactor the if block into `if (this.generation == generation)` instead of having an early return, but the code will become more nested.

Reply:
Ooh, `if (this.generation == generation)` might actually be nicer. I guess it does basically the same thing, but without making the reader think through a lint rule and whether it's helpful here or not. I'd like to here @gnprice's thoughts; I'll mark for his review.

Reply:
Yeah. The version in `main` uses `if (this.generation == generation)` here instead of our usual early return, and I think this lint rule was the main reason I went for that.

One extra layer of nesting is pretty tolerable, particularly as the indentation size is just 2 spaces.